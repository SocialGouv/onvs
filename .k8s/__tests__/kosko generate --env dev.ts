//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  process.env.HARBOR_PROJECT = "onvs";
  expect(
    await getEnvManifests("dev", "", {
      ...project("onvs").dev,
      KUBE_NAMESPACE: "onvs-171-master-dev2",
      RANCHER_PROJECT_ID: "c-bd7z2:p-ngv88",
    })
  ).toMatchSnapshot();
});
