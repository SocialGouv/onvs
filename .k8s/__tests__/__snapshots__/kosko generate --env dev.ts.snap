// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bd7z2:p-ngv88
    git/branch: master
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/onvs.git
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  labels:
    azure-pg-admin-user: onvs
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  name: onvs-85-master-dev42
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  labels:
    app: app
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  name: app
  namespace: onvs-85-master-dev42
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-onvs
        app.gitlab.com/env: master-dev42
        app.gitlab.com/env.name: master-dev42
      labels:
        app: app
        application: master-dev42-onvs
        owner: onvs
        team: onvs
        cert: wildcard
    spec:
      containers:
        - image: >-
            harbor.fabrique.social.gouv.fr/onvs/onvs-app:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: app
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 200m
              memory: 256Mi
            requests:
              cpu: 50m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: azure-pg-user-master-dev42
            - secretRef:
                name: app-sealed-secret
            - configMapRef:
                name: app-configmap
      initContainers:
        - env:
            - name: WAIT_FOR_RETRIES
              value: '24'
          envFrom:
            - secretRef:
                name: azure-pg-user-master-dev42
          image: ghcr.io/socialgouv/docker/wait-for-postgres:6.32.2
          imagePullPolicy: Always
          name: wait-for-postgres
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  name: app-sealed-secret
  namespace: onvs-85-master-dev42
  labels:
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
spec:
  encryptedData:
    MAIL_HOST: >-
      AgCmfO/0u0+1cNvvY+pam8GrklZY905zwc9fRWkYVV/NBJzhyzgpC9UuSoNjVl+h3zSqogxmBHi1weX1Xd/uwHX1TtwFjyJBGDglu9JmTE8ointnCLhahbdfL/St6wn5OkyXQ00nTfxUGg+7RI1uvesEPF7PbLz6KahbHDqPSQEqUoDoWtcsL3kf9oSqzmuBdpSiBGPivXbPPLoBCoKfBVzS799vS/v+Hdq1/KBQfW+bHrAgj4AbEZe5LMqZsDqDkCra7Br4wsKbm4u42fkEhyiU8oNUZita1+6Wy/CJRKJ58tiXfEhTs9gtdf++AcOFESYba9xIUIXqcCiuS3cHAaWw/87GDCvoU/oJL0/qyy6W0bFT+eI7GCHwbK6XUVEzOL4BPXVUAtuXoc0Hk5WQrrVVw+XnBX8jCUQWTjY1tINNTVyyHItq5PgNDy/7IoajketMohjraCHWJ+psUNh3g6PBBSFilNmFd9RwVInlSotS3La8sFTv+XSzCkWQFXTKSbn3ByAGCDdJvLrX8mHZNEGDEvjzdFaEGb3bD5SBlomci4z3J7c1/Pvd1uInxRC2wk3UltHGKRsp7nWcTbbeudChZQ5saxUXyD2S13ZrQTgu25g8/6baeE9Qpof5fvpGTaDCR4p6b3jR0EQzTIm/IFJC4XAOPM+GAdLbfjq3hTMvFVGwd59+MZCcJdoTN9cklQrcFdyLIEfkUgZgiC2dQXiZ
    MAIL_PORT: >-
      AgCBQppqHQCQy1TO8B/shjCcCx1+M4vxMBET18YY8AMtk+swYxx/Re+X7SEWWIx2MYobgedrwrmCtU22B5L89L/E1JpC4G6kpyo6WsXZMRZyeI1lxITMWP8Vk874V1xlXl+Z1LMFK7Glc7ONhge3bb8oXJ/QOHa4DVnVm1BZc00HYnJ6cblTY139e8QwRKiyVqVT+u+a6sPqcS9v8uSxGdFSMrj27nf7HwZrwuoNERJ3oMYPwicuBbn5RY6CsuqopAnlCBQnBthUu5hvSZVBVLATTQN4Ywt89+dMjs+3TVXHT9xeal1mkz7spyRVQFp2fFEazfqsOr3OqPuVj5XIpUkqRVSdZOsopLWjd38cjOM7p6I9+UvoH+ulQ7ln7qqOMfanFOPQnwoLpuZC7IZUWjdXJBlJdQhkfYos1C9CkVTCbqf+0FPGKk3YkBmINiYY6d6AMPgkKidTSA22exoEeJnCce/wc52yc8p/gr2S1KRdlq/J29xcLqZ5jB1QO4wsp2OUTncKR5eAeTUWFpSHLclsnPdzLghleZ+57JDRNBrsRv6P6IdMQDOss8dUzPvnUvm/1vcpYkYPh8aa2wyPkzcVF3+3xr+4NidJlq3LHHEsuHAbE9mpvmA9VLbTBwpXivUTSoChF60Jh43HS4IuS2AGC4hytzzGRHilDA/XpLOvZfO5OfuTT9utQWjrMDUGrtT8qNyl3w==
    MAIL_USERNAME: >-
      AgAnVPNQV6W3/hng42rhz0SdYFSvxgE8IclyeBICTSIbU9IgiI2YCMMwk/j0DwKoLgWx3nFPFCGfHv8tWfcSmfiaJq6fhEK+NCQbupiNiwttM+gQxQp7TYyaPSF81ZO2GlYPxdnFeih93njrcGvZEWdfyKo0V8hxPgx2ZYWjlfC8TCwHUZiwe/rDVqEvhFBpruZnNwwuoNiYmQXewZuuguFGxHM2SAJaABNcf2W7TNw60H0Vho7Vk0J9lWy/4hcHDuun+rrq6jkmJ9ktktBe/U3peDW6Ido0X4fCcnjohmkSRZh/o7nJu6UdI5J+pNlYprc2T+PkkZZEFo7+0GwRMzCYkWVFPmxX3EObtKWmkMeZcTRJAqLih4TKqanSr+IukQ/kJJwWnhtZmKrqUUDZ9MJwkvtZVeLkugPzDNb7McQQ2U/6bfMP39ThfeOSx2pwMX5qBGSqmQpUPuXeFm4MT450uMwB9bkqLSWVXQGYYliosbb/KGJajLbeTie7JbHz86/tbyGaDt4ycZ9icx/+IRaPdhvsa6CLbV8SybOogyApcqJ+o8AAtSCkx29fNs2UXuKSEK0HQe2XuYqCHSixL+l9GmjSteqcvPQk67DD4yjjTAizN9M2/4A7p6YxjRYn/+dUyBTn1p1Hf6WKt31auZaOwW4LxZVwUV2AeieATjMJ73h8yKTBmGtYq1hJcsXg/P1Hhr5GFIYK23H0dOGeSQ==
    MAIL_PASSWORD: >-
      AgBB/frhpbGeUuVWg9OoUY5veBMJR9W7lIgXKrLl3rGdFNsJa++M/AfJ7hyDCW14otKH6Chp5sNHxw6u+VvAplZC0HLTUXyt5Ax7E9lKUFAy+wF5MsOy8sutjkuIgeFwEm+o1E5RtcMNzFBJLegmYTaJyRJ9aPhPaVs8TdTFRlTEn0VBIF35HLbYgiNrdas1dLZyr6Xuw/HJYd0zQqiEZKOHmJ74R5+q+D4RBeknnwy0j9TnG5FPHknjuEsHg8zwUUGXGUoe/0Wk+0PW6tOxVfTMb9HFCpYgd9+BaFD8GI9/UfCRG9kdUtT7z6MLJGgL8SNzjgHuNLF/vtr3KmSf47qMh96qCDOC8rfjBWnNd91gEsJnWhvvrX6RjbWey62mnW+JYxaLgko32w89P2z3osRv4Oxv00C5YpMRAJdXkUWdnR/jLFlycOxtQ/f9q0m0+WnWaaC1V4F6jO02+b5wWUdXdEEd8leAkpN+/TweKMO9ABf/cQAjau+fiagWf5Rt3dLs8C+n3FadYxXLKMIZyVVuQOGSsRdo4Q6/klKhJmHesDygk4EJEADkOVPOAUesV0uOFmsULM11RC7wrA6IlcU5cUFxiOYBo9QOaEZTy+6SDEtQYttWFayMGaX4oE0YF8rDFhN84GqtMa9feoCVCP/ejd1Wnp/4DATLRezig/tZYxR96HDpdSBrFN0zQvA6FaT0AGdr1gA0WD2pF+zRoA==
    MAIL_TO: >-
      AgCTo9v4muXmw081Gir5zqGj/gWFs9uw6dgpAri6vRDlzx5h0qs9deXOSGHAuecF4n66pt4+EI+UcMrM/qObqs/Nu/gkS52GN/2dBnjILwmwGPclfsS+K4ZgL/TRtWQUGHfJ/l9+3B2MopT8hyImwCy4iKlpW/rml9kdCBuey0UfL5qdUBZHORp9wF/3e5YFDA9MAyK1ixu2noHoNwskcdDABitRkcpFtZtVkBjO6e30S07HA5HZbGppm/+QIl+kqaUqL/H05Gz0iXwqwYPBuMW0qwB9MUyz3jtphg9RiVyZ7EHCv1/7cZWhDazrNuuSUEhz001HkwWW3YsyQ8ilT6EhRGMrIhVoqXWH7fJBdt0MPKSdPE8F7X8IGcuJvRQLFpuNPvJNInjz+YeqcK4aQ8SzZTRM/4kNfNueTLrK+Kd6Yi9ld0UOUArY/Ubhda+pr1vRDykHjy39H38yoYXPmXPQ3yE9EFcMWyHOH+3iSjcmKR+3zJE+BXc9ujaERfCRMOp4vH2N9lSS12aLLOonwltahF0Yw5CjpUsz+RR69BfRNY4WsF6lAOjHhe40Gh8pAbaZmSO86si649fzVagw7GyZYYxnHQj7ChpLs7hKE/ofceIbHoG2Jh0mJYyEZkfS2VcwZx3xIXQxl1Yei2VytBRBGi9WPKuYm+IUVWtc3XG/SZ0X6TS7RcIQ/uoN8zHEair6IFuJ7QtaBLuYRtEE4DryRYNw1g==
    SECRET_COOKIE_PASSWORD: >-
      AgDFyz0Bkk8b6V+HfpkcsIrrX9f1fzlwoCnPyexWiG2uhEgzYcr9yola4ZO2LlyVek6Tlvz+gs68fyIY6seqLdk7R/gbufUNOR8fv5FC2Z14FntXns+1t//p1YTTB6c8h+UMrsqbajp+s9l5ewDYAl1mYrfZGOg2nVEBeMrN/fJfSTIrLc18tfQzO/HtVtkMj/PxIUJ/deTgEu2gjBdQyx1zrwB9C/7ByP0hWAAHUNU+fRWG2HMaFEDbETjERfE0FuJs5gUYPhq9MTmKf+pKIyw8cF9cUndydBmq/6et4d+VZLko6as8TXp+bgO0XZ3yBjKNAu6DYKuGacwcsw1xlJO/pK0qFRLe8n8GqF8EOOI3NRxAXr64N/QMjh2DZkhHcFg+AhHNGRj4YfakfHNm2E4R6CfbhtSvMAOCgUlFrbbUvhqepiCiDHy60WtAaSWGon55/89ocJE0yuv4HRfV98xddfnBt0TeciZuwZpRpxTppniXCOd4t1eElKU5iwe7EJq1PdQd3aeHzalaiTVm4+YfbJ3pGbP0UD8gbMExBrhDNny8yiDYi4zPudbkrOPM6RL5dInmKZ+OYwWhga7OjIZ6OjnyS0QU644+Al1cYco/uD/JDqxuh+bJZ+0ojRODLIjpODJj8RlEpDyuCBij4GFVUz1BSr7rwiGc7BApegSR0Hy/V7ZeuGRDdduCy1OAReTBgUgGB11MQCKcgKQyadul4kRJqk73sB0fKH9bq5P7n/Gd/lM=
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-onvs
        app.gitlab.com/env: master-dev42
        app.gitlab.com/env.name: master-dev42
      name: app-sealed-secret
      labels:
        application: master-dev42-onvs
        owner: onvs
        team: onvs
        cert: wildcard
    type: Opaque
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: app-configmap
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  labels:
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  namespace: onvs-85-master-dev42
data:
  NODE_ENV: production
  PORT: '3000'
  NEXT_PUBLIC_SENTRY_TOKEN: toto
  NEXT_PUBLIC_SENTRY_DSN: https://e70207cd7a2c4ecdb99849c212a964bf@sentry.fabrique.social.gouv.fr/40
  NEXT_PUBLIC_MATOMO_URL: https://matomo.fabrique.social.gouv.fr/
  NEXT_PUBLIC_MATOMO_SITE_ID: '25'
  POSTGRES_SSL: 'true'
  MAIL_WEBHOOK_TOKEN: superSecret
  MAIL_FROM: ONVS <noreply@fabrique.social.gouv.fr>
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: app
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  name: app
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  namespace: onvs-85-master-dev42
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: app
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  labels:
    app: app
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  name: app
  namespace: onvs-85-master-dev42
spec:
  rules:
    - host: master-dev42-onvs.dev42.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: app
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - master-dev42-onvs.dev42.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: batch/v1
kind: Job
spec:
  backoffLimit: 5
  template:
    spec:
      containers:
        - command:
            - create-db-user
          env:
            - name: NEW_DB_NAME
              value: autodevops_master-dev42
            - name: NEW_USER
              value: user_master-dev42
            - name: NEW_PASSWORD
              value: password_master-dev42
            - name: NEW_DB_EXTENSIONS
              value: hstore pgcrypto citext uuid-ossp
          envFrom:
            - secretRef:
                name: azure-pg-admin-user
          image: ghcr.io/socialgouv/docker/azure-db:6.32.2
          imagePullPolicy: IfNotPresent
          name: create-db-user
          resources:
            limits:
              cpu: 300m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 64Mi
      restartPolicy: Never
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-onvs
        app.gitlab.com/env: master-dev42
        app.gitlab.com/env.name: master-dev42
      labels:
        application: master-dev42-onvs
        owner: onvs
        team: onvs
        cert: wildcard
  ttlSecondsAfterFinished: 86400
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  labels:
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  name: create-db-job-0123456
  namespace: onvs-85-master-dev42
---
apiVersion: v1
kind: Secret
stringData:
  DATABASE_URL: >-
    postgresql://user_master-dev42%40onvsdevserver.postgres.database.azure.com:password_master-dev42@onvsdevserver.postgres.database.azure.com/autodevops_master-dev42?sslmode=require
  DB_URI: >-
    postgresql://user_master-dev42%40onvsdevserver.postgres.database.azure.com:password_master-dev42@onvsdevserver.postgres.database.azure.com/autodevops_master-dev42?sslmode=require
  HASURA_GRAPHQL_DATABASE_URL: >-
    postgresql://user_master-dev42%40onvsdevserver.postgres.database.azure.com:password_master-dev42@onvsdevserver.postgres.database.azure.com/autodevops_master-dev42?sslmode=require
  PGDATABASE: autodevops_master-dev42
  PGHOST: onvsdevserver.postgres.database.azure.com
  PGPASSWORD: password_master-dev42
  PGRST_DB_URI: >-
    postgresql://user_master-dev42%40onvsdevserver.postgres.database.azure.com:password_master-dev42@onvsdevserver.postgres.database.azure.com/autodevops_master-dev42?sslmode=require
  PGSSLMODE: require
  PGUSER: user_master-dev42@onvsdevserver.postgres.database.azure.com
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-onvs
    app.gitlab.com/env: master-dev42
    app.gitlab.com/env.name: master-dev42
  labels:
    application: master-dev42-onvs
    owner: onvs
    team: onvs
    cert: wildcard
  name: azure-pg-user-master-dev42
  namespace: onvs-85-master-dev42
"
`;
